src/main/java/uk/gov/companieshouse/web/accounts/controller/smallfull/TurnoverPolicyController.java

```diff
@@ -0,0 +1,93 @@
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import uk.gov.companieshouse.web.accounts.annotation.NextController;
+import uk.gov.companieshouse.web.accounts.annotation.PreviousController;
+import uk.gov.companieshouse.web.accounts.controller.BaseController;
+import uk.gov.companieshouse.web.accounts.exception.ServiceException;
+import uk.gov.companieshouse.web.accounts.model.smallfull.notes.accountingpolicies.TurnoverPolicy;
+import uk.gov.companieshouse.web.accounts.service.smallfull.TurnoverPolicyService;
+import uk.gov.companieshouse.web.accounts.util.Navigator;
+import uk.gov.companieshouse.web.accounts.validation.ValidationError;
+
+@Controller
+@NextController(ReviewController.class)
+@PreviousController(BasisOfPreparationController.class)
+@RequestMapping("/company/{companyNumber}/transaction/{transactionId}/company-accounts/{companyAccountsId}/small-full/turnover-policy")
+public class TurnoverPolicyController extends BaseController {
+
+    @Autowired
+    private TurnoverPolicyService turnoverPolicyService;
+
+    @GetMapping
+    public String getTurnoverPolicy(@PathVariable String companyNumber,
+        @PathVariable String transactionId,
+        @PathVariable String companyAccountsId,
+        Model model,
+        HttpServletRequest request) {
+
+        addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+        try {
+            model.addAttribute("turnoverPolicy",
+                turnoverPolicyService.getTurnOverPolicy(transactionId, companyAccountsId));
+
+        } catch (ServiceException e) {
+            LOGGER.errorRequest(request, e.getMessage(), e);
+            return ERROR_VIEW;
+        }
+
+        return getTemplateName();
+    }
+
+    @PostMapping
+    public String postTurnoverPolicy(@PathVariable String companyNumber,
+        @PathVariable String transactionId,
+        @PathVariable String companyAccountsId,
+        @ModelAttribute("turnoverPolicy") @Valid TurnoverPolicy turnoverPolicy,
+        BindingResult bindingResult,
+        Model model,
+        HttpServletRequest request) {
+
+        addBackPageAttributeToModel(model, companyNumber, transactionId, companyAccountsId);
+
+        if (bindingResult.hasErrors()) {
+            return getTemplateName();
+        }
+
+        try {
+            List<ValidationError> validationErrors = turnoverPolicyService
+                .postTurnoverPolicy(transactionId, companyAccountsId, turnoverPolicy);
```

Comment:
Can we rename this method `submitTurnoverPolicy`? The controller needn't know about the type of `CRUD` operation performed by the service, plus since validation is now performed in this method 'submission' feels like it covers the functionality of the method

Reply:
Amost all controllers are following this naming convention. I dont think this change is really needed. Please, let me know what you think.

Reply:
After speaking with Brad. We agreed not to rename this method.